apps/ybal/components/CardTransactorWrapper.tsx

```diff
@@ -0,0 +1,298 @@
+	const expectedOutFetcher = useCallback(async (args: [string, string, BigNumber]): Promise<BigNumber> => {
+		const [_inputToken, _outputToken, _amountIn] = args;
+		if (_amountIn.isZero()) {
+			return (Zero);
+		}
+
+		const	currentProvider = provider || getProvider(1);
+		if (_inputToken === YBAL_BALANCER_POOL_ADDRESS) {
+			// Direct deposit to vault from bal/yBal Curve LP Token to lp-yBal Vault
+			const	contract = new ethers.Contract(
+				LPYBAL_TOKEN_ADDRESS,
+				['function pricePerShare() public view returns (uint256)'],
+				currentProvider
+			);
+			try {
+				const	pps = formatBN(await contract.pricePerShare());
+				const	_expectedOut = _amountIn.mul(pps).div(ethers.constants.WeiPerEther);
+				return _expectedOut;
+			} catch (error) {
+				return (Zero);
+			}
+		} else {
+			// Zap in
+			const	contract = new ethers.Contract(
+				ZAP_YEARN_VE_CRV_ADDRESS,
+				['function calc_expected_out(address, address, uint256) public view returns (uint256)'],
+				currentProvider
+			);
+			try {
+				const	_expectedOut = formatBN(await contract.calc_expected_out(_inputToken, _outputToken, _amountIn));
+				return _expectedOut;
+			} catch (error) {
+				return (Zero);
+			}
+		}
+	}, [provider]);
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** SWR hook to get the expected out for a given in/out pair with a specific
+	** amount. This hook is called every 10s or when amount/in or out changes.
+	** Calls the expectedOutFetcher callback.
+	**************************************************************************/
+	const	{data: expectedOut} = useSWR(
+		isActive ? [
+			selectedOptionFrom.value,
+			selectedOptionTo.value,
+			amount.raw
+		] : null,
+		expectedOutFetcher,
+		{refreshInterval: 30000, shouldRetryOnError: false, revalidateOnFocus: false}
+	);
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** Approve the spending of token A by the corresponding ZAP contract to
+	** perform the swap.
+	**************************************************************************/
+	async function	onApproveFrom(): Promise<void> {
+		new Transaction(provider, approveERC20, set_txStatusApprove).populate(
+			toAddress(selectedOptionFrom.value),
+			selectedOptionFrom.zapVia,
+			ethers.constants.MaxUint256
+		).onSuccess(async (): Promise<void> => {
+			await refresh();
+		}).perform();
+	}
+
+	/* 🔵 - Yearn Finance ******************************************************
+	** BAL token require the allowance to be reset to 0 before being able to
+	** increase it. This function is called when the user wants to increase the
+	** allowance of the BAL token.
+	**************************************************************************/
+	async function	onIncreaseBalAllowance(): Promise<void> {
+		await new Transaction(provider, approveERC20, set_txStatusApprove).populate(
+			toAddress(selectedOptionFrom.value),
+			selectedOptionFrom.zapVia,
+			0
+		).perform();
```

Comment:
I am finding these chained function calls a bit hard to read, I'd rather have it like this
```suggestion
		const tx = new Transaction(provider, approveERC20, set_txStatusApprove);
		tx.populate(toAddress(selectedOptionFrom.value), selectedOptionFrom.zapVia, 0);
		await tx.perform();
```

Reply:
Perfect!